source/qdk_package/README.md

```diff
@@ -0,0 +1,186 @@
+```
+
+```python
+from qdk import azure_available, require
+
+if azure_available():
+    azure = require("azure")
+    # Example: azure.Workspace(...) etc., per azure-quantum docs
+```
+
+## Public API Surface
+
+Root-level symbols (kept intentionally small):
+
+| Symbol                | Description                                                                 |
+| --------------------- | --------------------------------------------------------------------------- |
+| `qsharp`              | Submodule re-export of the upstream `qsharp` package.                       |
+| `widgets_available()` | Boolean: is widget support (jupyter extra) installed?                       |
+| `azure_available()`   | Boolean: is the azure extra installed?                                      |
+| `qiskit_available()`  | Boolean: is the qiskit extra installed?                                     |
+| `require(name)`       | Retrieve a feature module (`"qsharp"`, `"widgets"`, `"azure"`, `"qiskit"`). |
+| (lifted utilities)    | Convenience re-exports from `qsharp` (see section below).                   |
+
+Submodules:
+
+- `qdk.qsharp` – direct passthrough to `qsharp` APIs.
+- `qdk.widgets` – only if `qsharp-widgets` installed (through `qdk[jupyter]`).
+- `qdk.azure` – only if `azure-quantum` installed.
+- `qdk.qiskit` – only if `qiskit` installed.
+- `qdk.estimator` – shim re-export of `qsharp.estimator` (always present if underlying `qsharp` provides it).
+- `qdk.openqasm` – shim re-export of `qsharp.openqasm` for OpenQASM integration.
+
+### Lifted utilities from `qsharp`
+
+For convenience, frequently-used helpers and types are available directly at the `qdk` root. Algorithm execution APIs (like `run` / `estimate`) remain under `qdk.qsharp`.
+
+| Symbol               | Type     | Origin                      | Description                                                         |
+| -------------------- | -------- | --------------------------- | ------------------------------------------------------------------- |
+| `code`               | module   | `qsharp.code`               | Define inline Q# snippets / code objects.                           |
+| `set_quantum_seed`   | function | `qsharp.set_quantum_seed`   | Deterministic seed for quantum randomness (simulators).             |
+| `set_classical_seed` | function | `qsharp.set_classical_seed` | Deterministic seed for classical host RNG.                          |
+| `dump_machine`       | function | `qsharp.dump_machine`       | Emit a structured dump of full quantum state (simulator dependent). |
+| `dump_circuit`       | function | `qsharp.dump_circuit`       | Produce a circuit representation / diagram (when supported).        |
+| `Result`             | class    | `qsharp.Result`             | Measurement result token.                                           |
+| `TargetProfile`      | class    | `qsharp.TargetProfile`      | Target capability / profile descriptor.                             |
```

Comment:
`TargetProfile` and `init` should be present in the same (sub)module, as they are commonly used together. It will be annoying to have to `from qdk.qsharp import init` and then have to `from qdk import TargetProfile` so that you can call `init` with the profile specified.

Reply:
Take a look through https://github.com/microsoft/qdk/blob/main/samples/notebooks/openqasm.ipynb if that helps.